core/src/main/java/com/google/instrumentation/trace/SpanBuilder.java

```diff
@@ -0,0 +1,277 @@
+ *     }
+ *     // "span" will be ended here.
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>This is a simple example where all the work is being done within a single scope and a single
+ * thread and the Context is automatically propagated:
+ *
+ * <pre>{@code
+ * class MyClass {
+ *   private static final Tracer tracer = Tracer.getTracer();
+ *   void doWork {
+ *     // Create a Span as a child of the current Span.
+ *     try (NonThrowingCloseable ss = tracer.spanBuilder("MyChildSpan").startScopedSpan()) {
+ *       tracer.getCurrentSpan().addAnnotation("my annotation");
+ *       doSomeWork();  // Here the new span is in the current Context.
+ *     }
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>There might be cases where you do not perform all the work inside one static scope and the
+ * Context is automatically propagated:
+ *
+ * <pre>{@code
+ * class MyWorkClass {
+ *   private static final Tracer tracer = Tracer.getTracer();
+ *   Span mySpan;
+ *
+ *   MyWorkClass() {
+ *     // Create a Span as a child of the current Span.
+ *     mySpan = tracer.spanBuilder("MyChildSpan").startSpan();
+ *   }
+ *
+ *   void doInitialWork {
+ *     try (NonThrowingCloseable ws = tracer.withSpan(mySpan)) {
+ *       tracer.getCurrentSpan().addAnnotation("my annotation in initial work");
+ *       doSomeInitialWork(); // Here the new span is in the current Context.
+ *     }
+ *   }
+ *
+ *   void doFinalWork() {
+ *     try (NonThrowingCloseable ws = tracer.withSpan(mySpan)) {
+ *       tracer.getCurrentSpan().addAnnotation("my annotation in final work");
+ *       doSomeFinalWork(); // Here the new span is in the current Context.
+ *     }
+ *     // IMPORTANT: DO NOT forget to ended the Span here.
+ *     mySpan.end();
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>If your Java version is less than Java SE 7, see {@link SpanBuilder#startSpan} and {@link
+ * SpanBuilder#startScopedSpan} for usage examples.
+ */
+public final class SpanBuilder {
+  private final SpanFactory spanFactory;
+  private final ContextSpanHandler contextSpanHandler;
+  private SpanContext parentSpanContext;
+  private boolean hasRemoteParent;
+  private final String name;
+  private final StartSpanOptions startSpanOption = new StartSpanOptions();
+
+  SpanBuilder(
+      SpanFactory spanFactory,
+      ContextSpanHandler contextSpanHandler,
+      SpanContext parentSpanContext,
+      boolean hasRemoteParent,
+      String name) {
+    this.parentSpanContext = parentSpanContext;
+    this.hasRemoteParent = hasRemoteParent;
+    this.name = name;
+    this.spanFactory = spanFactory;
+    this.contextSpanHandler = contextSpanHandler;
+  }
+
+  /**
+   * Sets the start time for the {@link Span}.
+   *
+   * @param startTime The start time for the {@code Span}. If {@code null} is used, then the current
+   *     system time at the point at which {@link #startSpan} is called will be used.
+   * @return this.
+   */
+  public SpanBuilder setStartTime(@Nullable Timestamp startTime) {
```

Comment:
personal taste, but I usually leave off "set" prefix, and don't add fillers like "return this"